js-miniapp-sample/src/services/uuid/actions.js

```diff
@@ -0,0 +1,17 @@
+// @flow
+import { getUUIDFromMobileSdk } from "./../../js_sdk";
+import { SET_UUID } from "./types";
+
+type GetUUIDAction = { type: String, payload: string };
+
+type UUIDAction = GetUUIDAction;
+
+const setUUID = (): GetUUIDAction => {
+  return {
+    type: SET_UUID,
+    payload: getUUIDFromMobileSdk(),
```

Comment:
Getting a UUID is going to be a side effect (the definition is loose, but I understand it as any observable interaction with the outside world).

Redux follow functional programming concepts. For example, reducers are pure functions. Since actions are inputs into reducers (that describe _what_ changed), they need to be pure as well. Since we're introducing an impure calculation, we'll need to update this action creator to be a thunk that returns a function, which is a common technique for delaying the evaluation until it's called.

I suggest changing the `setUUID` function to:

```javascript
const setUUID = () => {
  return dispatch => {
    dispatch({
      type: SET_UUID,
      payload: getUUIDFromMobileSdk(),
    });
  };
};
```

And then installing `redux-thunk` and setting it up as a middleware to handle thunks in `src/services/store.js`:

```javascript
import {createStore, applyMiddleware} from "redux";
import thunkMiddleware from "redux-thunk";

import reducers from "./reducers";

export default createStore(reducers, applyMiddleware(thunkMiddleware));
```


More info: https://redux.js.org/advanced/async-actions#async-action-creators